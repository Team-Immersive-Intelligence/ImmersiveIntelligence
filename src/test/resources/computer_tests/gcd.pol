use ADVANCED_ARITHMETIC
use COMPARATOR
use LOGIC

integer a = 198
integer b = 24

type @a
type @b


exec "gcd"
type @c

end

;Euclid's Algorithm - calculates greatest common divisor
;uses @a and @b as parameters
;returns @c

mark gcd
integer a = modulo @a @b
swap a b

type @a
type @b

if ! (equal @b 0)
    goto "gcd"

copy a c
end