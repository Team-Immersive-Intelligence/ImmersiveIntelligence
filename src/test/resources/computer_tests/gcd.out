[POL] 198
[POL] 24
[POL] 24
[POL] 6
[POL] 6
[POL] 0
[POL] 6
